src/managers/RoleManager.js

```diff
@@ -182,12 +191,20 @@ class RoleManager extends CachedManager {
       });
     }
 
+    let icon = data.icon;
+    if (icon) {
+      const guildEmojiURL = this.guild.emojis.resolve(icon)?.url;
+      icon = guildEmojiURL ? await DataResolver.resolveImage(guildEmojiURL) : await DataResolver.resolveImage(icon);
```

Comment:
ditto
```suggestion
      const emojiURL = this.guild.emojis.resolve(icon)?.url;
      icon = emojiURL ? await DataResolver.resolveImage(emojiURL) : await DataResolver.resolveImage(icon);
```

Reply:
change `this.guild.emojis` to `this.client.emojis` like above

Reply:
On second thought, `Client#emojis` seems really expensive and it's gonna be called every time there is an icon param irrespective of whether it's an EmojiResolvable or not. Do we really want to add this when (I assume) Discord client too only shows emojis in the guild?

Reply:
The discord client does not show any guild emojis all, this is just a convenience. And how is it expensive?

Reply:
the client does show guild emojis, and client.emojis iterates through all guilds to get all of their emojis on every access

Reply:
The difference would be barely noticeable if you’re just using get though

Reply:
Made the behaviour similar to Discord client by only allowing the same guild emojis. Added warnings about this behaviour along with how to use emojis from other guilds as an icon

Reply:
Sounds good to me